Python/bytecodes.c

```diff
@@ -2182,7 +2182,14 @@ dummy_func(
             JUMPBY(1-oparg);
             #if ENABLE_SPECIALIZATION
             here[1].cache += (1 << OPTIMIZER_BITS_IN_COUNTER);
-            if (here[1].cache > tstate->interp->optimizer_backedge_threshold) {
+            if (here[1].cache > tstate->interp->optimizer_backedge_threshold &&
+                // Double-check that the opcode isn't instrumented or something:
+                here->op.code == JUMP_BACKWARD &&
+                // _PyOptimizer_BackEdge is going to change frame->prev_instr,
+                // which breaks line event calculations:
+                next_instr->op.code != INSTRUMENTED_LINE
```

Comment:
What about EXTENDED_ARG?

Reply:
And in general, we want to be able to form traces through large loops. The most likely reason our tarket would be an `EXTENDED_ARG` is if we're about to jump back to the top of a big for-loop (`EXTENDED_ARG; FOR_ITER; ...; JUMP_BACKWARD`).